### PR TITLE
[Controller] Added board config for custom controller STeMCell (updated)

### PR DESCRIPTION
**Pull request by sadekbaroudi**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by sadekbaroudi** [comment]
This pull request is based on the previous pull request by @Mariappan ( https://github.com/qmk/qmk_firmware/pull/16287 )

I have updated it to latest develop, and added a commit to support the changes to pindefs. I have tested on a few fingerpunch keyboards using the stemcell.

**Review by keyboard-magpie — APPROVED** [review]
I have been using this with a stemcell for the last week, without issue. RGB features and i2c pointing device included. EEPROM features appear to be working appropriately although have not tested VIA only lighting options.

The PR, I believe, addresses all the issues raised in the "stuck" PR #16287 , but I am unsure of the etiquette surrounding merging this ahead of the stuck PR.

**Comment by sadekbaroudi** [comment]
Oh, just to explicitly note, some of the objections in the last PR were addressed by creating this feature: https://github.com/qmk/qmk_firmware/pull/16598

I am also leveraging this so we don't need to use the STMC=yes and can instead use the standardized CONVERT_TO=stemcell

File: platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h
Comment (zvecr) [inline platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h:7]: This doesnt belong in pin defs
```suggestion
```

File: platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h
Comment (zvecr) [inline platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h:54]: This doesnt belong in pin defs
```suggestion
```

File: platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h
Comment (zvecr) [inline platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h:1]:
```suggestion
// Copyright 2022 QMK
// SPDX-License-Identifier: GPL-2.0-or-later

#pragma once

// Left side (front)
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Some of the changes still fall under the same issues as <https://github.com/qmk/qmk_firmware/pull/17711>, where specifying the full board/chibos config files wants to be avoided.

For EEPROM, it would be to avoid `STEMCELL_tinyuf2` linker script and be solved more generically for `BOOTLOADER = tinyuf2`

**Comment by sadekbaroudi** [comment]
> Some of the changes still fall under the same issues as #17711, where specifying the full board/chibos config files wants to be avoided.
> 
> For EEPROM, it would be to avoid `STEMCELL_tinyuf2` linker script and be solved more generically for `BOOTLOADER = tinyuf2`

For reference, I assume this conversation is relevant, so linking it here:
https://github.com/qmk/qmk_firmware/pull/16287#discussion_r822234320

I don't know enough about how this part of the codebase works, so deferring to @Mariappan to chime in.

As a side note, I'm happy to contribute and/or fix myself, but some more detailed guidance would be very helpful!

**Comment by sigprof** [comment]
The EEPROM emulation should work with `BOOTLOADER = tinyuf2` if you use
```makefile
EEPROM_DRIVER = wear_leveling
WEAR_LEVELING_DRIVER = legacy
```
(`WEAR_LEVELING_DRIVER = embedded_flash` currently does not support F4x1 chips.)

`STEMCELL_tinyuf2.ld` does not seem to be needed — it's effectively the same as `STM32F401xC_tinyuf2.ld` (and if the real chip is F411, the RAM size is wrong), except it also defines flash regions for the bootloader and EEPROM backing store, but those regions are not actually used anywhere.

Board config should be inherited from a ChibiOS definition, like in `GENERIC_STM32_F411XE`; the only problem is `mcuconf.h`, which is also copied in full in existing converters like `QMK_PM2040`, so this one may need to remain as is for now.

**Comment by sadekbaroudi** [comment]
@Mariappan has started working on https://github.com/qmk/qmk_firmware/pull/16287 again, so I'll stop using this PR, and defer to that one. If it's okay, I'll keep this one open until that one gets merged in. But for now, this one is on pause.

Please let me know if anyone has any objections to this.

**Comment by zvecr** [comment]
replaced by https://github.com/qmk/qmk_firmware/pull/16287

**Pull request closed without merging**
